mongomock/database.py

```diff
@@ -63,10 +65,39 @@ def collection_names(self, include_system_collections=True, session=None):
 
         return self.list_collection_names(session=session)
 
-    def list_collection_names(self, session=None):
+    def list_collection_names(self, filter=None, session=None):
+        """filter: only name field type with eq,ne or regex operator
+
+        session: not supported
+        supported operator are $eq, $ne, $regex
+        """
+
+        allowed_operators = ['$regex', '$eq', '$ne']
+        filed_name = 'name'
+
+        def verify_supported_op(keys):
+            for key in keys:
+                if key not in allowed_operators:
+                    raise NotImplementedError(
+                        'Mongomock list collection names filter'
+                        ' operators are {0}'.format(allowed_operators))
+
         if session:
             raise NotImplementedError('Mongomock does not handle sessions yet')
 
+        if filter:
+            if not filter.get('name'):
+                raise NotImplementedError('Mongomock list collection names support name type only')
```

Comment:
Again mention somewhere that the filter might be valid for pymongo but the operation is not implemented yet in mongomock.

Reply:
done